internal/daemon/api_changes.go

```diff
@@ -165,18 +165,52 @@ func v1GetChanges(c *Command, r *http.Request, _ *userState) Response {
 }
 
 func v1GetChange(c *Command, r *http.Request, _ *userState) Response {
-	chID := muxVars(r)["id"]
-	state := c.d.overlord.State()
-	state.Lock()
-	defer state.Unlock()
-	chg := state.Change(chID)
+	changeID := muxVars(r)["id"]
+	st := c.d.overlord.State()
+	st.Lock()
+	defer st.Unlock()
+	chg := st.Change(changeID)
 	if chg == nil {
-		return statusNotFound("cannot find change with id %q", chID)
+		return statusNotFound("cannot find change with id %q", changeID)
 	}
 
 	return SyncResponse(change2changeInfo(chg))
 }
 
+func v1GetChangeWait(c *Command, r *http.Request, _ *userState) Response {
+	changeID := muxVars(r)["id"]
+	st := c.d.overlord.State()
+	st.Lock()
+	change := st.Change(changeID)
+	st.Unlock()
+	if change == nil {
+		return statusNotFound("cannot find change with id %q", changeID)
+	}
+
+	timeoutStr := r.URL.Query().Get("timeout")
+	if timeoutStr != "" {
+		// Timeout specified, wait till change is ready or timeout occurs,
+		// whichever is first.
+		timeout, err := time.ParseDuration(timeoutStr)
+		if err != nil {
+			return statusBadRequest("invalid timeout %q: %v", timeoutStr, err)
+		}
+		timer := time.NewTimer(timeout)
+		select {
+		case <-change.Ready():
+			timer.Stop() // change ready, release timer resources
+		case <-timer.C:
```

Comment:
add select clause for select on r.Context().Done()

Reply:
Thanks - fixed!